runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java

```diff
@@ -381,8 +381,21 @@ public void onPageLoadStarted(XWalkViewInternal view, String url) {
     @XWalkAPI
     public void onPageLoadStopped(XWalkViewInternal view, String url, LoadStatusInternal status) {
     }
-
-    private boolean onJsAlert(XWalkViewInternal view, String url, String message,
+    
+    /**
+     * Tell the client to display an alert dialog to the user.
+     * WARN: Please avoid to override this API and onJavascriptModalDialog API in the 
+     *       same subclass to avoid unexpected behavior. If cannot avoid, Please 
```

Comment:
Just don't let the user do that, don't be ambiguous. Use assert expression to forbid it